front/src/common/Map/WarpedMap/SimulationWarpedMap.tsx

```diff
@@ -0,0 +1,322 @@
+        (position.headDistanceAlong / pathLength) * transformedPathLength;
+      transformedTrain.tailDistanceAlong =
+        (position.tailDistanceAlong / pathLength) * transformedPathLength;
+
+      return { ...transformedTrain, train: trainsIndex[position.trainId] };
+    });
+  }, [
+    itineraryState,
+    simulation,
+    timePosition,
+    positionValues,
+    selectedTrain,
+    allowancesSettings,
+    state,
+  ]);
+
+  /**
+   * This effect handles loading the simulation path, and retrieve the warping function:
+   */
+  useEffect(() => {
+    setState({ type: 'loading' });
+    getPath({ id: pathfindingID })
+      .then(({ data, isError, error }) => {
+        if (isError) {
+          setState({ type: 'error', message: error as string });
+        } else if (!data?.geographic?.coordinates) {
+          setState({ type: 'error', message: 'No coordinates' });
+        } else {
+          const path = lineString(data?.geographic?.coordinates as Position[]);
+          const pathBBox = bbox(path) as BBox2d;
+          const { regularBBox, transform } = getWarping(path);
+
+          setState({ type: 'pathLoaded', path, pathBBox, regularBBox, transform });
+        }
+      })
+      .catch((error) => setState({ type: 'error', message: error }));
+  }, [pathfindingID]);
+
+  /**
+   * This effect tries to gradually improve the quality of the OSRD data.
+   * Initially, all OSRD entities are with "simplified" geometries, due to the
+   * fact that they are loaded directly using an unzoomed map.
+   */
+  useEffect(() => {
+    if (state.type !== 'dataLoaded') return;
+
+    getImprovedOSRDData(infraID as number, state.osrd).then((betterFeatures) => {
+      if (!isEmpty(betterFeatures)) {
+        const betterTransformedFeatures = mapValues(betterFeatures, state.transform);
+        const newTransformedOSRDData = mapValues(state.osrd, (collection: FeatureCollection) => ({
+          ...collection,
+          features: collection.features.map(
+            (feature) => betterTransformedFeatures[feature.properties?.id] || feature
+          ),
+        }));
+        setState({ ...state, osrd: newTransformedOSRDData });
+      }
+    });
+  }, [state]);
+
+  return (
+    <div
+      className="warped-map position-relative d-flex flex-row"
+      style={{ width: collapsed ? 0 : WIDTH }}
+    >
+      {state.type === 'pathLoaded' && (
+        <DataLoader
+          bbox={state.pathBBox}
+          layers={layers}
+          getGeoJSONs={(osrdData, osmData) => {
+            console.time(TIME_LABEL);
+            const transformed = {
+              osm: omitBy(
+                mapValues(osmData, (collection) => state.transform(collection)),
+                isNil
+              ) as DataStatePayload['osm'],
+              osrd: omitBy(
+                mapValues(osrdData, (collection: FeatureCollection) => state.transform(collection)),
+                isNil
+              ) as DataStatePayload['osrd'],
+            };
```

Comment:
Suggestion : create a helper function like this:
```typescript
const transformDataStatePayload = (
  data: Partial<Record<string, FeatureCollection<Geometry, GeoJsonProperties>>>,
  transform: WarpingFunction
) =>
  omitBy(
    mapValues(data, (collection) => (collection ? transform(collection) : null)),
    isNil
  );
```

and then use it to make the code clearer
```suggestion
            const transformed = {
              osm: transformDataStatePayload(osmData, state.transform) as DataStatePayload['osm'],
              osrd: transformDataStatePayload(
                osrdData,
                state.transform
              ) as DataStatePayload['osrd'],
            };
```